x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/store/CompositeRolesStore.java

```diff
@@ -239,7 +239,7 @@ public void getRoles(User user, Authentication authentication, ActionListener<Ro
                     },
                     roleActionListener::onFailure
                 ));
-            } else {
+            } else {  // TODO: the else branch is for bwc and can be deleted in 9.0
```

Comment:
I don't think this is true even in 9.0 - we don't have anything in place that would automatically rewrite saved authentication objects from 7.x watcher jobs.

Reply:
You are right. I [realised it](https://github.com/elastic/elasticsearch/pull/58156#issuecomment-644732410) a while ago, then I just forgot about it again. Thanks for reminding me.